src/components/Products/ProductList/ProductList.component.jsx

```diff
@@ -27,19 +35,48 @@ const ProductList = () => {
     setPage,
     totalPages,
   } = usePagination({
-    contentPerPage: 16,
+    contentPerPage: 12,
     count: products.length,
   });
 
   // setting up filter logic
   const [filterArray, setFilterArray] = useState([]);
 
+  useEffect(() => {
+    if (slugCategory !== '') {
+      setFilterArray([...filterArray, slugCategory]);
+      console.log(filterArray);
+    }
+  }, []);
+
   const categoryFilter = (id) => {
     filterArray.includes(id)
       ? setFilterArray(filterArray.filter((x) => x !== id))
-      : setFilterArray([...filterArray, id]);
+      : //setFilterArray(filterArray.push(id));
+        setFilterArray(...filterArray, id);
+    console.log(filterArray);
+    const filtered = products.filter((product) =>
+      product.data.category.slug.includes(filterArray)
+    );
+    console.log(filterArray, filtered);
```

Comment:
There are bunch of these console logs, there are fine for development, but we should always remove them as they could leak information. You can config add a config in eslint to warn you about this and add a linter step before every push. Let me know if you need help in case you want to set this up.